routes/registration.js

```diff
@@ -0,0 +1,116 @@
+require('dotenv').config();
+const express = require('express');
+const jwt = require('jsonwebtoken');
+const router = express.Router();
+const { check, validationResult } = require('express-validator');
+const bcrypt = require('bcrypt');
+const User = require('../models/User');
+const nodemailer = require('nodemailer');
+const mailGun = require('nodemailer-mailgun-transport');
+
+// VALIDATION INITIALIZATION
+const validations = [
+  check('first_name').trim().isLength({ min: 5 }).escape().withMessage('name is required'),
+  check('last_name').trim().isLength({ min: 5 }).escape().withMessage('name is required'),
+  check('email').trim().isEmail().normalizeEmail().withMessage('A valid email address is required'),
+  check('username').trim().escape(),
+  check('password')
+    .trim()
+    .isLength({ min: 8 })
+    .escape()
+    .withMessage('Password is required and should not be less than 8 characters'),
+  check('age').trim().escape().isInt().withMessage('age is required'),
+  check('gender').trim().escape(),
+];
+
+module.exports = () => {
+  router.get('/', (req, res) => {
+    res.render('api/user/register');
+  });
+
+  router.post('/', validations, async (req, res) => {
+    //VALIDATE USER BEFORE WE SAVE
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).send(errors.array());
+    }
+
+    //CHECK IF EMAIL EXIST IN THE DATABASE
+    const emailExist = await User.findOne({ email: req.body.email });
+    if (emailExist) {
+      return res.status(400).json({ message: 'Email already exist, try another one' });
+    }
+
+    //HASH USER PASSWORD USING BCRYPT
+    const salt = await bcrypt.genSalt(10);
+    const hashedPassword = await bcrypt.hash(req.body.password, salt);
```

Comment:
this can be in your model save middleware
https://mongoosejs.com/docs/4.x/docs/middleware.html